synapse/lib/base.py

```diff
@@ -302,11 +362,60 @@ def off(self, evnt, func):
         if self.finievt is None:
             self.finievt = asyncio.Event(loop=self.loop)
 
-        try:
-            await asyncio.wait_for(self.finievt.wait(), timeout, loop=self.loop)
-        except asyncio.TimeoutError:
-            return None
-        return True
+        return await s_coro.event_wait(self.finievt, timeout)
+
+    def schedCoro(self, coro):
+        '''
+        Schedules a free-running coroutine to run on this base's event loop.  Kills the coroutine if Base is fini'd.
+        It does not pend on coroutine completion.
+
+        Precondition:
+            This function is *not* threadsafe and must be run on the Base's event loop
+
+        Returns:
+            An asyncio.Task
+
+        '''
+        assert s_threads.iden() == self.tid
+
+        task = self.loop.create_task(coro)
+
+        def taskDone(task):
+            self._active_tasks.remove(task)
+            try:
+                task.result()
+            except asyncio.CancelledError:
+                pass
+            except Exception:
+                logger.exception('Task scheduled through Base.schedCoro raised exception')
+
+        self._active_tasks.add(task)
+        task.add_done_callback(taskDone)
+
+        return task
+
+    def schedCoroSafe(self, coro):
```

Comment:
Do we want to return the `task` here so a caller can wait for the result (if they so desire) ?

Reply:
ok